### PR TITLE
[WIP] Run script generator: Added a script builder for golang code

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

**Comment by kichalla** [comment]
Sent out this PR just to get any initial feedback before I proceed to making changes everywhere

File: src/startupscriptgenerator/src/common/outputHelper.go [hunk 1]
```diff
@@ -23,27 +22,25 @@ func AppendScriptToExtractZippedOutput(scriptBuilder *strings.Builder, appDir st
 	}
 
 	if FileExists(fullZipFilePath) {
-		scriptBuilder.WriteString("\n")
-		scriptBuilder.WriteString("echo Extracting '" + consts.CompressedOutputFileName + "' contents...\n")
-		scriptBuilder.WriteString("cd \"" + intermediateDir + "\"\n")
-		scriptBuilder.WriteString("tar -xzf \"" + consts.CompressedOutputFileName + "\"\n")
-		scriptBuilder.WriteString("echo Done.\n")
+		scriptBuilder.AppendEmptyLine()
```
Comment (kichalla) [inline src/startupscriptgenerator/src/common/outputHelper.go]: What do you think of this API?

File: src/startupscriptgenerator/src/common/outputHelper.go [hunk 2]
```diff
@@ -23,27 +22,25 @@ func AppendScriptToExtractZippedOutput(scriptBuilder *strings.Builder, appDir st
...
-	scriptBuilder.WriteString("    rm -rf \"$destDir\"\n")
-	scriptBuilder.WriteString("    echo Done.\n")
-	scriptBuilder.WriteString("fi\n")
-	scriptBuilder.WriteString("mkdir -p \"$destDir\"\n")
-	scriptBuilder.WriteString("echo \"Copying content from '$srcDir' to directory '$destDir'...\"\n")
-	scriptBuilder.WriteString("cp -rf \"$srcDir\"/* \"$destDir\"\n")
-	scriptBuilder.WriteString("echo Done.\n")
+func AppendScriptToCopyToDir(scriptBuilder *ScriptBuilder, srcDir string, destDir string) {
+	scriptBuilder.AppendEmptyLine()
+	scriptBuilder.AppendLine("if [ -d \"" + destDir + "\" ]; then")
```
Comment (kichalla) [inline src/startupscriptgenerator/src/common/outputHelper.go]: Few complexities related to blocks like `if`:
1. Expressing conditions: condition1 or condition2, condition1 and condition 2
2. Indentation of statements below this if condition. Ideally we might want to use api like `ChangeToDirectory` under this condition too, but the problems are: a). we would loose indentation when generating the script b). Even if we can fix the indentation (keep track of these kind of blocks using a stack) in the generated script, when reading this code in our code base, have indentation in golang code is also needed

File: src/startupscriptgenerator/src/common/outputHelper.go (hunk 1 above)
Comment (dorfire) [reply src/startupscriptgenerator/src/common/outputHelper.go]: I like the idea! would probably go for even shorter names (`builder.Chdir`, `builder.ExtractArchive`).
great initiative 👍

File: src/startupscriptgenerator/src/common/scriptBuilder.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// --------------------------------------------------------------------------------------------
+
+package common
+
+import (
+	"strings"
+)
+
+type ScriptBuilder struct {
+	scriptBuilder strings.Builder
```
Comment (dorfire) [inline src/startupscriptgenerator/src/common/scriptBuilder.go]: member `scriptBuilder` in type `ScriptBuilder` seems redundant - maybe just `script`?

File: src/startupscriptgenerator/src/common/outputHelper.go (hunk 2 above)
Comment (gpcastro) [reply src/startupscriptgenerator/src/common/outputHelper.go]: one thing we could do is to add a `scriptBuilder.AddIfBlock(condition: string)` that returns a `ifBlock` object, in a fluent design. We could do this:
```
scriptBuilder.AddIfBlock("-d \"" + destDir + "\"")
    .appendLine(...)
    .end()
```

Comment (dorfire) [reply src/startupscriptgenerator/src/common/outputHelper.go]: The aforementioned IfBlock object itself could probably be another `ScriptBuilder` instance

**Pull request closed without merging**